core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1020,6 +1020,17 @@ class ReplicaManager(val config: KafkaConfig,
             exception = None)
         }
       } catch {
+        case e@ (_: OffsetOutOfRangeExceptionWithOffsetValues) =>
+          LogReadResult(info = FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY),
+              highWatermark = Log.UnknownOffset,
+              leaderLogStartOffset = e.getLogStartOffset,
```

Comment:
We should add test cases which verify that out of range errors include these offsets. I suggest looking at `FetchRequestTest`.

Reply:
Added testOffsetOutOfRangeExceptionWithOffsetValues()